src/config/config.go

```diff
@@ -0,0 +1,58 @@
+package config
+
+import (
+	"log"
+
+	"github.com/spf13/viper"
+)
+
+var Global config
+
+type config struct {
+	Env           string `mapstructure:"GO_ENV"`
+	Version       int    `mapstructure:"VERSION"`
+	Port          int    `mapstructure:"PORT"`
+	Cookie        string `mapstructure:"COOKIE_NAME"`
+	FirestoreCred string `mapstructure:"FIRESTORE_CREDENTIALS"`
+	CorsUrl       string `mapstructure:"CORS_URL"`
+}
+
+var environmentToFileMaps = map[string]string{
+	"prod":    "production",
+	"test":    "testing",
+	"default": "default",
+	"local":   "local",
+}
+
+func getConfigFile(env string) string {
+	var configFile = environmentToFileMaps[env]
+	return configFile
+}
+
+func loadConfig(env string) (config config, err error) {
+	var configFile = getConfigFile(env)
+
+	viper.AddConfigPath("src/config")
+
+	viper.SetConfigName("default")
+	viper.ReadInConfig()
+
+	viper.SetConfigName(configFile)
+	viper.MergeInConfig()
+
+	viper.AutomaticEnv()
+
+	err = viper.Unmarshal(&config)
+	return

```

Comment:
When are we reading configs from local?

Reply:
When the env is blank we read from the local
you can see it here:
https://github.com/Real-Dev-Squad/website-learn-backend/pull/10/files/8d3aa05f20afee0693f905c091d801564e08cfab#diff-73f70add140b12d299684fe58706a57775a3e7dacc4116045276dfa2e5c8ca78R49-R51

Reply:
So when in development mode, we should read from local if its present, 

When you run in development, will you add credentials in development.toml which is tracked?

Or always add and then remove again and add again?